### DISCUSSION TITLE
Problem with Github OAuth

### DISCUSSION THREAD

**Discussion by ELatestark** [WeblateOrg/weblate#12446] [Q&A]
I've added self-signed ssl certs to the app and now it operates behind https.
![Снимок экрана от 2024-09-08 17-21-48](https://github.com/user-attachments/assets/60c01b26-d457-4520-9d5f-892c3924cb4f)
When I try to login to GHE it redirects me to our GHE domain to authenticate:
![Снимок экрана от 2024-09-08 17-23-13](https://github.com/user-attachments/assets/391618eb-1309-4434-86e6-765d16fe5ef3)
Then it tells me that I'm going to be redirected to the app, but it never happens.
![Снимок экрана от 2024-09-08 17-29-46](https://github.com/user-attachments/assets/c70f59a6-6da7-4fe2-9348-c91737b1aec8)
Auth webpage loads endlessly, but never loads and then I got redirected to main page and this error shows up:
![Снимок экрана от 2024-09-08 17-34-07](https://github.com/user-attachments/assets/05737fe8-0662-413b-af38-e36eb9c81e71)
In Network tab I see the request that doesnt work:
![Снимок экрана от 2024-09-08 17-36-38](https://github.com/user-attachments/assets/aec77033-f0e3-44eb-a982-2db6868bb5bc)
My docker-compose.override.yml:
![Снимок экрана от 2024-09-08 17-06-11](https://github.com/user-attachments/assets/4b4f7e15-ea34-4fd7-bbfd-f7d516a4dd34)
OAuth config:
![Снимок экрана от 2024-09-08 17-49-18](https://github.com/user-attachments/assets/42730d37-41ab-441d-aca2-7500b686b4e8)

I've seen [similar post ](https://github.com/WeblateOrg/weblate/issues/5173), but WEBLATE_ENABLE_HTTPS is already set.
Tried to unset WEBLATE_ENABLE_HTTPS, but immediately got "redirect_uri_mismatch" and turned it back :D

No "uri mismatch" errors or anything, I'm stuck.
Requesting assistance!

**Comment by nijel** [comment]
Check Weblate logs, it should tell you more info. Can Weblate reach the GHE server?

**Reply by ELatestark** [reply]
Hm, yes, it can.
![Снимок экрана от 2024-09-09 15-50-03](https://github.com/user-attachments/assets/46a9070d-1b7d-41d9-a03a-d992b0cd9872)
I restarted my docker compose with same configuration and now endless loading disappeared. What a magic.
Now I'm getting uri_mismatch:
![Снимок экрана от 2024-09-09 15-52-19](https://github.com/user-attachments/assets/10eb7626-66e0-47d3-8fc0-27cac905ce7a)
Configuration is the same I mentioned above, what should I look for?

**Reply by ELatestark** [reply]
Found out if I change OAuth config to http it starts working and letting me log in:
![Снимок экрана от 2024-09-09 16-47-24](https://github.com/user-attachments/assets/b0ac12c9-4e79-4eaf-ae53-4e4365b7478e)
Is that expected? Can I still have https Authorization callback URL and be able to log in?

**Reply by nijel** [reply]
It's not expected. Did you look into Weblate logs?
